test/common.py

```diff
@@ -2270,7 +2270,7 @@ def compile_btest(self, filename, args, reporting=Reporting.FULL):
       if reporting == Reporting.FULL:
         # If C reporting (i.e. the REPORT_RESULT macro) is required we
         # also include report_result.c and force-include report_result.h
-        self.run_process([EMCC, '-c', '-I' + TEST_ROOT,
+        self.run_process([EMCC, '-c', '-fPIC', '-I' + TEST_ROOT,
```

Comment:
Hmm.. I wonder why this is needed now and not for any other existing dynamic linking browser tests.    I will take a look.

Reply:
Instead of doing this can you do `+ args` at the end of this command line?

Reply:
(you may also need to add `-Wno-unused`)

Reply:
No problem.

Reply:
@sbc100 Looks like `args` doesn't get passed into this call:
https://github.com/emscripten-core/emscripten/blob/16559db15997f6de648c9107e26033043f8a5313/test/common.py#L2273-L2275

Reply:
Yes, I suggest just adding `+ args` after the `+ self.get_emcc_args(compile_only=True)`

Reply:
Some of the args passed to `compile_btest()` will interfere with that. I've added detection for `-fPIC` and am adding ONLY that flag if its present in `args`:

https://github.com/emscripten-core/emscripten/blob/feab2d60ef886ed6b07ae6cf187d312c2d2b65ad/test/common.py#L2275

Reply:
Which argument interfere?  Is it your test or some other test where the interference occurs?

Reply:
My test fails if I just do `+ args`:

```
Test suites:
['test_browser']
Running test_browser: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
[Browser harness server on process 152289]
common:INFO: No EMTEST_BROWSER set. Defaulting to `google-chrome`
common:INFO: Launching browser: ['google-chrome']

Running the browser tests. Make sure the browser allows popups from localhost.

test_fetch_polyfill_shared_lib (test_browser.browser.test_fetch_polyfill_shared_lib) ... Opening in existing browser session.
emcc: error: linker setting ignored during compilation: 'EXIT_RUNTIME' [-Wunused-command-line-argument] [-Werror]
None
None
FAIL
[Browser harness server terminated]

======================================================================
FAIL: test_fetch_polyfill_shared_lib (test_browser.browser.test_fetch_polyfill_shared_lib)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 1550, in run_process
    return shared.run_process(cmd, check=check, **args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sean/projects/php-wasm/emscripten/tools/shared.py", line 130, in run_process
    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/sean/projects/php-wasm/emscripten/emcc', '-c', '-I/home/sean/projects/php-wasm/emscripten/test', '-DEMTEST_PORT_NUMBER=8888', '/home/sean/projects/php-wasm/emscripten/test/report_result.c', '-Wclosure', '-Werror', '-Wno-limited-postlink-optimizations', '-Wno-pointer-sign', '-Wno-int-conversion', '-fPIC', 'library.so', '-sMAIN_MODULE=2', '-sEXIT_RUNTIME', '-o', 'a.out.html', '--pre-js', '/home/sean/projects/php-wasm/emscripten/test/browser_reporting.js']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sean/projects/php-wasm/emscripten/test/test_browser.py", line 5585, in test_fetch_polyfill_shared_lib
    test([], expect_fail=True)
  File "/home/sean/projects/php-wasm/emscripten/test/test_browser.py", line 5577, in test
    self.compile_btest('main.c', ['-fPIC', 'library.so', '-sMAIN_MODULE=2', '-sEXIT_RUNTIME', '-o', 'a.out.html'] + args)
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 2273, in compile_btest
    self.run_process([EMCC, '-c', '-I' + TEST_ROOT,
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 1555, in run_process
    self.fail(f'subprocess exited with non-zero return code({e.returncode}): `{shared.shlex_join(cmd)}`')
AssertionError: subprocess exited with non-zero return code(1): `/home/sean/projects/php-wasm/emscripten/emcc -c -I/home/sean/projects/php-wasm/emscripten/test -DEMTEST_PORT_NUMBER=8888 /home/sean/projects/php-wasm/emscripten/test/report_result.c -Wclosure -Werror -Wno-limited-postlink-optimizations -Wno-pointer-sign -Wno-int-conversion -fPIC library.so -sMAIN_MODULE=2 -sEXIT_RUNTIME -o a.out.html --pre-js /home/sean/projects/php-wasm/emscripten/test/browser_reporting.js`

----------------------------------------------------------------------
Ran 1 test in 0.364s

FAILED (failures=1)
```

Reply:
You can add `-Wno-unused-command-line-argument` too.. since its reasonable to ignore linker-only flags here.

Reply:
With: `'-Wno-unused-command-line-argument`:

```
Test suites:
['test_browser']
Running test_browser: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
[Browser harness server on process 156229]
common:INFO: No EMTEST_BROWSER set. Defaulting to `google-chrome`
common:INFO: Launching browser: ['google-chrome']

Running the browser tests. Make sure the browser allows popups from localhost.

test_fetch_polyfill_shared_lib (test_browser.browser.test_fetch_polyfill_shared_lib) ... Opening in existing browser session.
emcc: error: report_result.o: No such file or directory ("report_result.o" was expected to be an input file, based on the commandline arguments provided)
None
None
FAIL
[Browser harness server terminated]

======================================================================
FAIL: test_fetch_polyfill_shared_lib (test_browser.browser.test_fetch_polyfill_shared_lib)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 1550, in run_process
    return shared.run_process(cmd, check=check, **args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sean/projects/php-wasm/emscripten/tools/shared.py", line 130, in run_process
    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/sean/projects/php-wasm/emscripten/emcc', 'main.c', '-sNO_DEFAULT_TO_CXX', '-Wclosure', '-Werror', '-Wno-limited-postlink-optimizations', '-Wno-pointer-sign', '-Wno-int-conversion', '-fPIC', '-Wno-unused-command-line-argument', 'library.so', '-sMAIN_MODULE=2', '-sEXIT_RUNTIME', '-o', 'a.out.html', '--pre-js', '/home/sean/projects/php-wasm/emscripten/test/browser_reporting.js', 'report_result.o', '-include', '/home/sean/projects/php-wasm/emscripten/test/report_result.h']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sean/projects/php-wasm/emscripten/test/test_browser.py", line 5585, in test_fetch_polyfill_shared_lib
    test([], expect_fail=True)
  File "/home/sean/projects/php-wasm/emscripten/test/test_browser.py", line 5577, in test
    self.compile_btest('main.c', ['-fPIC', '-Wno-unused-command-line-argument', 'library.so', '-sMAIN_MODULE=2', '-sEXIT_RUNTIME', '-o', 'a.out.html'] + args)
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 2281, in compile_btest
    self.run_process([compiler_for(filename), filename] + self.get_emcc_args() + args)
  File "/home/sean/projects/php-wasm/emscripten/test/common.py", line 1555, in run_process
    self.fail(f'subprocess exited with non-zero return code({e.returncode}): `{shared.shlex_join(cmd)}`')
AssertionError: subprocess exited with non-zero return code(1): `/home/sean/projects/php-wasm/emscripten/emcc main.c -sNO_DEFAULT_TO_CXX -Wclosure -Werror -Wno-limited-postlink-optimizations -Wno-pointer-sign -Wno-int-conversion -fPIC -Wno-unused-command-line-argument library.so -sMAIN_MODULE=2 -sEXIT_RUNTIME -o a.out.html --pre-js /home/sean/projects/php-wasm/emscripten/test/browser_reporting.js report_result.o -include /home/sean/projects/php-wasm/emscripten/test/report_result.h`

----------------------------------------------------------------------
Ran 1 test in 0.485s

FAILED (failures=1)
```

Reply:
Ah thats kind of a shame.. the args include the `-o .. ` flags.   I guess you `-fPIC` approach is fine.